packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1131,48 +1190,103 @@ class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigation
     }
   }
 
+  void _handleSearchFieldStatusChanged(AnimationStatus status) {
+    switch (status) {
+      case AnimationStatus.completed:
+      case AnimationStatus.dismissed:
+        setState(() {});
+      case AnimationStatus.forward:
+        expanded = false;
+      case AnimationStatus.reverse:
+        expanded = true;
+    }
+  }
+
+  void _onSearchFieldTap() {
+    if (widget.onSearchActiveChanged != null) {
+      widget.onSearchActiveChanged!(expanded);
+    }
+    _animationController.toggle();
+    _fadeController.toggle();
+  }
+
   @override
   Widget build(BuildContext context) {
     final _NavigationBarStaticComponents components = _NavigationBarStaticComponents(
       keys: keys,
       route: ModalRoute.of(context),
-      userLeading: widget.leading,
+      userLeading:
+          widget.leading != null ? Visibility(visible: expanded, child: widget.leading!) : null,
       automaticallyImplyLeading: widget.automaticallyImplyLeading,
       automaticallyImplyTitle: widget.automaticallyImplyTitle,
       previousPageTitle: widget.previousPageTitle,
-      userMiddle: widget.middle,
-      userTrailing: widget.trailing,
-      userLargeTitle: widget.largeTitle,
+      userMiddle: _animationController.isAnimating ? const Text('') : widget.middle,
+      userTrailing:
+          widget.trailing != null ? Visibility(visible: expanded, child: widget.trailing!) : null,
+      userLargeTitle:
+          _animationController.isAnimating
```

Comment:
Do we need to conditionally insert the FadeTransition? This might cause the tree to rebuild unnecessarily when this widget is removed from the tree after animating for example.